app/models/host/discovered.rb

```diff
@@ -55,6 +55,18 @@ def self.import_host_and_facts facts
     # and save (interfaces are created via puppet parser extension)
     host.save(:validate => false) if host.new_record?
     state = host.import_facts(facts)
+
+    # lock the host into discovery via PXE, if feature is enabled in settings
+    if Foreman::Cast.to_bool(Setting['discovery_lock']) and !Setting['discovery_lock_template'].blank?
+      begin
+        @host = ::ForemanDiscovery::HostConverter.to_managed(host)
+        template = @host.unattended_render(::ProvisioningTemplate.find_by_name(Setting['discovery_lock_template']).template)
+        Subnet.subnet_for(host.ip).tftp_proxy.set host.mac, :pxeconfig => template
+      rescue Exception => e
+        Rails.logger.error "Could not set tftp_proxy from Proxy: #{e}"
+      end
+    end
```

Comment:
Ok looks good, since you use `unattended_renderer`, you can also render a snippet. Can you make a patch to https://github.com/theforeman/community-templates pulling out this block https://github.com/theforeman/community-templates/blob/develop/pxe/PXELinux_default.erb#L20-L24 into separate snippet, which you can use here?